docs/user/ppl/limitations/limitations.rst

```diff
@@ -9,13 +8,79 @@ Limitations
    :local:
    :depth: 2
 
+Inconsistent Field Types across indices
+=======================================
+
+* If the same field has different types across indices (e.g., ``field`` is a ``string`` in one index and an ``integer`` in another), PPL uses the first encountered type. Other types will be ignored during query execution.
```

Comment:
> PPL uses the first encountered type

QQ:
1. Is this deterministic or not?
2. Is this rule applied to object field merging logic below recursively?